docs/go/testing.md

```diff
@@ -183,3 +183,59 @@ function matches the signature of the original Activity function.
 
 Since this can be an entire function, there is no limitation as to what we can do here. In this
 example, we assert that the `value` param has the same content as the value param we passed to the Workflow.
+
+## Queries
+
+`TestWorkflowEnvironment` instances have a [`QueryWorkflow()` method](https://pkg.go.dev/go.temporal.io/temporal/internal#TestWorkflowEnvironment.QueryWorkflow) that lets you query the currently running Workflow.
+For example, suppose you have a Workflow that lets you query the progress of a long running task as shown below.
+
+```go
+func ProgressWorkflow(ctx workflow.Context, percent int) error {
+	logger := workflow.GetLogger(ctx)
+
+	err := workflow.SetQueryHandler(ctx, "getProgress", func(input []byte) (int, error) {
+		return percent, nil
+	})
+	if err != nil {
+		logger.Info("SetQueryHandler failed.", "Error", err)
+		return err
+	}
+
+	for percent = 0; percent<100; percent++ {
```

Comment:
Wondering why you need this loop. I think you can just change the value before returning and just query it in your test after workflow execution, or you can set it and wait for a signal in your test. 
This way your test depends on this 10 seconds + 2 milliseconds thing which can fail on some slow machines imo.

Reply:
I was trying to get an example of a workflow that has multiple intermediate states, not just 3 states like this example: https://github.com/temporalio/samples-go/blob/master/query/query_workflow.go. Think of it like a progress bar for an async operation, where the progress bar increments by 1% every second.

Reply:
Agreed, you don't want to rely on time for the test to pass. However, I like the idea of the progress loop, it's a fun example for query.

The other way to fix this is to change the test not to rely on a certain amount of time passing, but to simply keep querying in a loop (usually with a small sleep) until the value reaches 100. 

It's generally a good approach for this sort of thing. Don't wait the time you expect, but wait until the condition you expect is fulfilled (and, if time matters, which it doesn't really here, assert that reaching that condition didn't take too long)

Reply:
edit: Yeah, I forgot these were time-skipping tests. The concern isn't really important in these ones.